alicloud/resource_alicloud_sag_client_user.go

```diff
@@ -0,0 +1,158 @@
+package alicloud
+
+import (
+	"fmt"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/smartag"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudSagClientUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudSagClientUserCreate,
+		Read:   resourceAlicloudSagClientUserRead,
+		Update: resourceAlicloudSagClientUserUpdate,
+		Delete: resourceAlicloudSagClientUserDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"sag_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"bandwidth": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"user_mail": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"client_ip": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.SingleIP(),
+			},
+			"user_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(2, 128),
+			},
+			"password": {
```

Comment:
If one resource supports `password`, it should also need to support kms_password and you can refer to: https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_db_account.go#L44